mill

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env sh
+
+# This is a wrapper script, that automatically download mill from GitHub release pages
+# You can give the required mill version with --mill-version parameter
+# If no version is given, it falls back to the value of DEFAULT_MILL_VERSION
+
+DEFAULT_MILL_VERSION=0.2.7-10-db8525
+
+set -e
+
+if [ "x$1" = "x--mill-version" ] ; then
```

Comment:
My Bash is a bit rusty, but why did you add the `x` prefix before the LHS and RHS of the comparison?

Reply:
Also, to be clear, this `--mill-version` param must be the first argument, before any other parameters, and before `--interactive`/`-i`? If so we can give a nice error message if the user passes in `--mill-version` in another position, similar to what we do for `--interactive`/`-i`, by creating a dummy command-line argument in Mill's parser and error-ing out if that argument is provided. See https://github.com/lihaoyi/mill/blob/master/main/src/MillMain.scala#L94

Reply:
The "x$i" thing is an old habit. Can't tell you anymore, why that was needed, but 20 years ago, almost all scripts looked like this. Could be some prevention of some unwanted executions and the `test` tool, who knows?

Too keep the script simple, I parse that option only when it's the first argument. I'm a bit uncomfortable with an dummy `--mill-version` option in mill. It could also check if the given version is in fact the runtime version, as a safety measure. Then there would be no need to remove it.

Reply:
After a short web-search I found older/other shells could not compare an empty string properly, and some couldn't handle a `!` or `(` in `$1`. And someone said Solaris 10 still has issues.